### ISSUE TITLE
April 10 - 14 diffenator2 review

### ISSUE THREAD

**Issue by chrissimpkins** [googlefonts/googlesans-flex#504]
Thread for communication about the diffenator2 review of the fb-wip2 branch vs. GS Flex v1.001 release as of sources due to be delivered by David and Santiago on April 10.

**Comment by sannorozco** [comment]
Hallo @m4rc1e,

Sources are ready on _fb-wip2_ branch, the designspace we’re using is **GSFv2.00DBTBS.designspace**.

Let us know, cheers!

**Comment by m4rc1e** [comment]
Hey all,

Before I can actually review any of the fonts, @sannorozco needs to do the following:

- You need to rebase so your branch has the same commits as the current main branch
- There's going to be a lot of merge conflicts when you do rebase. Please rebase carefully. You've done some crazy things like deleting all the github actions workflows
- No need to check in the `fonts/` dir. This is built by the ci
- Please rename `GSFv2.00DBTBS.designspace` to `GoogleSansFlex.designspace`. We should be working on the same files, not creating new ones if we don't have to. This makes diffing much easier
- There should be one set of sources. Please delete the 5 odd wip designspaces you've added. There's nothing wrong with doing this while testing, I just wouldn't check them into git

**Comment by m4rc1e** [comment]
There's no fvar instances either. Please add the same instances as we had before.

```XML
(venv) Marcs-Air-2:~ marcfoley$ cat /Users/marcfoley/Downloads/fb-review/GoogleSansFlex\[GRAD\,ROND\,opsz\,wdth\,wght\].ttx 
<?xml version="1.0" encoding="UTF-8"?>
<ttFont sfntVersion="\x00\x01\x00\x00" ttLibVersion="4.38">

  <fvar>

    <!-- Weight -->
    <Axis>
      <AxisTag>wght</AxisTag>
      <Flags>0x0</Flags>
      <MinValue>1.0</MinValue>
      <DefaultValue>400.0</DefaultValue>
      <MaxValue>1000.0</MaxValue>
      <AxisNameID>258</AxisNameID>
    </Axis>

    <!-- Width -->
    <Axis>
      <AxisTag>wdth</AxisTag>
      <Flags>0x0</Flags>
      <MinValue>25.0</MinValue>
      <DefaultValue>100.0</DefaultValue>
      <MaxValue>151.0</MaxValue>
      <AxisNameID>259</AxisNameID>
    </Axis>

    <!-- Optical Size -->
    <Axis>
      <AxisTag>opsz</AxisTag>
      <Flags>0x0</Flags>
      <MinValue>6.0</MinValue>
      <DefaultValue>18.0</DefaultValue>
      <MaxValue>144.0</MaxValue>
      <AxisNameID>260</AxisNameID>
    </Axis>

    <!-- grade -->
    <Axis>
      <AxisTag>GRAD</AxisTag>
      <Flags>0x0</Flags>
      <MinValue>-50.0</MinValue>
      <DefaultValue>0.0</DefaultValue>
      <MaxValue>50.0</MaxValue>
      <AxisNameID>261</AxisNameID>
    </Axis>

    <!-- round -->
    <Axis>
      <AxisTag>ROND</AxisTag>
      <Flags>0x0</Flags>
      <MinValue>0.0</MinValue>
      <DefaultValue>0.0</DefaultValue>
      <MaxValue>100.0</MaxValue>
      <AxisNameID>262</AxisNameID>
    </Axis>
  </fvar>

</ttFont>
```

**Comment by m4rc1e** [comment]
I've finally hacked together something which builds so we get diffs:

https://github.com/googlefonts/googlesans-flex/actions/runs/4666534804

**Comment by m4rc1e** [comment]
Proportions have changed a lot for existing styles. This will definitely cause reflow.


![diffbrowsers_text-Windows-10-10 0 20348-SP0_chrome_112 0 5615 50](https://user-images.githubusercontent.com/7525512/231139687-72dcf57f-b728-4770-b541-cba48db79ecd.gif)


I'll get some design diffs in a sec

**Comment by m4rc1e** [comment]
![output](https://user-images.githubusercontent.com/7525512/231143090-98734d7e-797e-4a59-b625-5f7bd3428a0b.gif)

**Comment by m4rc1e** [comment]
Executive summary:

- Thin capitals appear narrower (this is probably the biggest cause of reflow)
- Thin, Regular space and nbspace are narrower
- punctuation is looser
- superiors or numr figures have different vertical alignment

cc @chrissimpkins @MariannaPaszkowska @davelab6

**Comment by davelab6** [comment]
The screenshots only cover default width 100, weights 100 400 700, and opsz active at px size.... 48pt = what px?

**Comment by m4rc1e** [comment]
correct. Diffs only cover the existing fvar instances.

I can check more locations in the designspace if you want? 

Ideally, It would be good for us to establish which coordinates are worth checking? if we do a cross product, we'll end up with too many diffs.

**Comment by davelab6** [comment]
I propose to only diff the 56 masters in the [fb-wip-2 branch designspace file](https://github.com/googlefonts/googlesans-flex/blob/fede56425f32a987625faf1751c1a54f40eff468/sources/regular/GoogleSansFlex.designspace) plus important intermediate locations on only the 4 axes in the main branch, opsz wght width ROND, eg wght 100 700, wdth 125, opsz 24, 45, 57 (from https://m3.material.io/styles/typography/type-scale-tokens)

**Comment by m4rc1e** [comment]
I'm still implementing the features in diffenator2. However, I've made some cross product diffs which can be found [here](https://drive.google.com/drive/folders/10WGzsbuvstuC2g1etknUrBjMi-1E6-cS?usp=share_link)

Main things I've noticed:

- Condensed styles are more compressed
- Expanded styles are slightly more compressed. I guess this is happening because M and N are narrower
- A is narrower in normal widths

I can take a deeper look tomorrow as well.

**Comment by chrissimpkins** [comment]
The diff comparisons were run on the v1.2 release candidates well after the April 2023 reviews that were documented here, and Marianna found these diffs to be acceptable.  We released and shipped the v1.2 files.  I think that this can be closed.
